keras_nlp/tokenizers/word_piece_tokenizer_trainer.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Trainer for Word Piece Tokenizer."""
+
+import tensorflow as tf
+import tensorflow_text as tf_text
+from tensorflow_text.tools.wordpiece_vocab import (
+    wordpiece_tokenizer_learner_lib as learner,
+)
+
+from keras_nlp.tokenizers.word_piece_tokenizer import PUNCTUATION_REGEX
+from keras_nlp.tokenizers.word_piece_tokenizer import (
+    WHITESPACE_AND_PUNCTUATION_REGEX,
+)
+
+
+def preprocess(text, lowercase, strip_accents, split):
```

Comment:
Oh sorry I meant let's use the same function for here and the word piece tokenizer itself. All of this is exactly equivalent with the tokenizer equivalent right right? You could call this function right after the convert to tensor line in the tokenizer call graph. This should live with the word piece tokenizer, and imported here.

Let's call also this pretokenize not preprocess.